code/src/components/ProductDetails.js

```diff
@@ -0,0 +1,156 @@
+`
+const PieChartLabel = styled.p`
+  font-size: 16px;
+`
+const IngredientsButton = styled.button`
+  background: transparent;
+  border: none;
+  font-size: 18px;
+`
+
+export const ProductDetails = () => {
+  const [showIngredients, setShowIngredients] = useState(false)
+  const params = useParams()
+  const history = useHistory()
+  const dispatch = useDispatch()
+  const selectedItem = useSelector((state) => state.nutrition.productDetails[params.product])
+
+  useEffect(() => {
+    if (selectedItem) {
+      return;
+    } else {
+      dispatch(fetchAndStore(params.product))
+    }
+  }, [])
+  
+  return (
+    <section>
+      {!selectedItem && <Loader />}
+      {selectedItem &&
+      <>
+        <Header>
+          <BackButton onClick={() => history.push('/')}><FontAwesomeIcon icon={faChevronCircleLeft} /></BackButton>
+          <p>{selectedItem.product.product_name}</p>
+        </Header>
+        <Container>
+          <PieChartContainer>
+            <PieChart
+              totalValue={selectedItem.product.nutriments.carbohydrates + selectedItem.product.nutriments.fat + selectedItem.product.nutriments.salt + selectedItem.product.nutriments.proteins + selectedItem.product.nutriments.fiber}
+              startAngle={0}
+              animate={true}
+              data={[
+                { title: 'Fat', value: selectedItem.product.nutriments.fat ? selectedItem.product.nutriments.fat : 0, color: '#692236' },
+                { title: 'Protein', value: selectedItem.product.nutriments.proteins ? selectedItem.product.nutriments.proteins : 0, color: '#e18634' },
+                { title: 'Carbs', value: selectedItem.product.nutriments.carbohydrates ? selectedItem.product.nutriments.carbohydrates : 0, color: '#be3d3b' },
+                { title: 'Salt', value: selectedItem.product.nutriments.salt ? selectedItem.product.nutriments.salt : 0, color: '#470202' },
+                { title: 'Fibre', value: selectedItem.product.nutriments.fiber ? selectedItem.product.nutriments.fiber : 0, color: '#9e5414' }
+              ]} />
+            </PieChartContainer>
+            <PieChartDetails>
+              <PieChartItem>
+                <PieChartSwatch color="#692236" />
+                <PieChartLabel>= fat</PieChartLabel>
+              </PieChartItem>
+              <PieChartItem>
+                <PieChartSwatch color="#e18634" />
+                <PieChartLabel>= protein</PieChartLabel>
+              </PieChartItem>
+              <PieChartItem>
+                <PieChartSwatch color="#be3d3b" />
+                <PieChartLabel>= carbohydrates</PieChartLabel>
+              </PieChartItem>
+              <PieChartItem>
+                <PieChartSwatch color="#470202" />
+                <PieChartLabel>= salt</PieChartLabel>
+              </PieChartItem>
+              <PieChartItem>
+                <PieChartSwatch color="#9e5414" />
+                <PieChartLabel>= fibre</PieChartLabel>
+              </PieChartItem>
```

Comment:
These colors and labels look surprisingly  familiar 🤔 
Any synergies between this and the data creation code?